### PR TITLE
Add reducers and actions

### PR DESCRIPTION
**Pull request by mershark**
For this task i implemented the following:
- Created a Redux store to centralize app data storage.
- Developed "booksSlice.js" to handle book data and actions.
- Set up "categoriesSlice.js" for category data management.
- Combined reducers in "store.js" for unified data handling.

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+const initialState = {
+  categories: [],
+  status: 'Under construction',
+};
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {},
+});
```
Comment (julie-ify) [inline src/redux/categories/categoriesSlice.js]: - [x] Kindly set up the categories reducer to return 'Under construction'. Here is an example.

  ```
  export const categoriesSlice = createSlice({
    name: "categories",
    initialState,
    reducers: {
      checkStatus: (state, action) => {
        state.categories =
          action.payload === "Under construction"
            ? "Under construction"
            : state.categories
      },
    },
  })
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @mershark,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (mershark) [reply src/redux/categories/categoriesSlice.js]: Alright, thank you.

**Review by usorfaitheloho — APPROVED** [review]
Hi @julie-ify  , 
# STATUS: APPROVED! 🥇 💯 

done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**